src/Cli/dotnet/commands/dotnet-check-update/BundleOutputFormatter.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Deployment.DotNet.Releases;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.NativeWrapper;
+using System.Linq;
+
+namespace Microsoft.DotNet.Tools.CheckUpdate
+{
+    internal class BundleOutputFormatter
+    {
+        protected ProductCollection _productCollection;
+
+        protected readonly IProductCollectionProvider _productCollectionProvider;
+
+        protected readonly IReporter _reporter;
+
+        public BundleOutputFormatter(
+            ProductCollection productCollection,
+            IProductCollectionProvider productCollectionProvider,
+            IReporter reporter)
+        {
+            _productCollection = productCollection;
+            _productCollectionProvider = productCollectionProvider;
+            _reporter = reporter;
+        }
+
+        protected bool BundleIsMaintenance(INetBundleInfo bundle)
+        {
+            return _productCollection
+                .First(product => product.ProductVersion.Equals($"{bundle.Version.Major}.{bundle.Version.Minor}"))
+                .SupportPhase.Equals(SupportPhase.Maintenance);
+        }
+
+        protected bool BundleIsEndOfLife(INetBundleInfo bundle)
```

Comment:
I'd suggest maybe using the IsOutOfSupport method from the Product class as that checks the EOL date. There can be a lag between when a product reaches its EOL date, but the support phase information is not yet updated in the json

Reply:
It looks like IsOutOfSupport is true for versions that are EOL or in maintenance, right? I think we want to treat these as separate, so we might have to continue to use the SupportPhase.

Reply:
It looks at the SupportPhase first (https://github.com/dotnet/deployment-tools/blob/16d870968ed84f38a7834b93103db126248f607f/src/Microsoft.Deployment.DotNet.Releases/src/Product.cs#L185) before checking the EndOfLifeDate.

Typically a product will move into Maintenance a few months before it becomes EOL. But, if the data is not updated to set it to EOL in time, the EndOfLifeDate is checked as a fallback.

So, if something is set to Maintenance and the EOL date hasn't been reached, it will return false, but once the EOL date is reached, it will return true, regardless of the support phase